src/coreclr/jit/gentree.cpp

```diff
@@ -23320,7 +23320,7 @@ GenTree* Compiler::gtNewSimdShuffleNode(
 #if defined(TARGET_XARCH)
     uint8_t  control   = 0;
     bool     crossLane = false;
-    bool     needsZero = varTypeIsSmallInt(simdBaseType) && (simdSize != 64);
+    bool     needsZero = (simdSize != 32) && (simdSize != 64);
```

Comment:
Would be simpler as:
```suggestion
    bool     needsZero = (simdSize <= 16);
```

Reply:
Done